File: Algorithmique appliquée/chapitre 1/python/exercice5.py
# -*-coding:utf-8 -*
# Programme exercice 5
# calcul du produit de puissances de 2
# date : 15/09/2016


import os
import math


produit = 1

# saisie de la puissance maximale
n = int(input("entrez la valeur max = "))

# calcul et affichage de la somme des puissances de 2
for k in range(1, n+1) :
    produit *= math.pow(2, k)
print("Le produit des 2 puissance k = ", produit)


os.system("pause")

File: Algorithmique appliquée/chapitre 1/cpp/exercice5.cpp
/*  programme produitPuissance(exercice5)
*   but: calcul du produit de puissance de 2
*   date: 15/09/2016
*/
#include <iostream>
#include <math.h>
using namespace std;

int main(int argc, char *argv[]){

    unsigned int k,n, produit =0;

    cout << "entrez la valeur max= ";
    cin >> n;

    for(k=1;k <= n ;k++){
        produit *= pow(2,k);
    }
    cout << "Le produit des 2 puissance k = " << produit;
    return 0;
}
